flutter/lib/src/navigation/display_strategy_evaluator.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+@internal
+class DisplayStrategyEvaluator {
+  static final DisplayStrategyEvaluator _instance =
+      DisplayStrategyEvaluator._internal();
+
+  factory DisplayStrategyEvaluator() {
+    return _instance;
+  }
```

Comment:
`l`: Is this the way to define a singleton in Dart? If yes, why do you define this as a singleton? Singletons are global state and global state is a bummer for testability.